nm: Use UUID for controller and parent

cathay4t opened the pull request:
When both user space only interface and kernel interface holding the
same name, NetworkManager might the user space interface for
controller/parent which lead to error as NetworkManager failed to find
kernel interface index for it.

Using UUID instead of interface name  in `NM.RemoteConnection.props.master`
, `NM.SettingVlan.props.parent` and etc could solve this problem.

Integration test cases included.

cathay4t commented:
This is workaround for NetworkManager bug https://bugzilla.redhat.com/show_bug.cgi?id=1914139

ffmancera commented on libnmstate/nm/active_connection.py:
```diff
@@ -336,10 +336,9 @@ def _activation_progress_check(self):
-            self._ctx, self._iface_name, self._iface_type
-        )
-        if self._nm_dev:
+        nm_dev = get_nm_dev(self._ctx, self._iface_name, self._iface_type)
+        if nm_dev:
+            self._nm_dev = nm_dev
```
I think this change is not related with this commit.

cathay4t replied:
Moved to https://github.com/nmstate/nmstate/pull/1486

ffmancera replied:
Thank you!

ffmancera commented on tests/integration/ovs_test.py:
```diff
@@ -766,3 +740,100 @@ def test_expect_failure_when_create_ovs_interface_without_bridge():
+
+
+def test_create_mac_vlan_over_ovs_iface_with_use_same_name_as_bridge(
+    ovs_bridge1_with_internal_port_same_name,
+):
+    desired_state = {
+        Interface.KEY: [
+            {
+                Interface.NAME: "mac_vlan101",
+                Interface.TYPE: InterfaceType.MAC_VLAN,
+                MacVlan.CONFIG_SUBTREE: {
+                    MacVlan.BASE_IFACE: BRIDGE1,
+                    MacVlan.MODE: MacVlan.Mode.PASSTHRU,
+                    MacVlan.PROMISCUOUS: False,
+                },
+            }
+        ]
+    }
+    libnmstate.apply(desired_state)
+    assertlib.assert_state_match(desired_state)
+
+
+def test_create_mac_vlan_over_ovs_iface_with_use_same_name_as_bridge(
+    ovs_bridge1_with_internal_port_same_name,
+):
+    desired_state = {
+        Interface.KEY: [
+            {
+                Interface.NAME: "mac_vlan0",
+                Interface.TYPE: InterfaceType.MAC_VLAN,
+                MacVlan.CONFIG_SUBTREE: {
+                    MacVlan.BASE_IFACE: BRIDGE1,
+                    MacVlan.MODE: MacVlan.Mode.PASSTHRU,
+                    MacVlan.PROMISCUOUS: False,
+                },
+            }
+        ]
+    }
+    libnmstate.apply(desired_state)
+    assertlib.assert_state_match(desired_state)
```
This test is duplicated.

ffmancera reviewed (changes requested):
LGTM, just a couple of nits. Thanks!

ffmancera reviewed (approved):
LGTM, thanks!

cathay4t committed:
nm: Use UUID for controller and parent

When both user space only interface and kernel interface holding the
same name, NetworkManager might the user space interface for
controller/parent which lead to error as NetworkManager failed to find
kernel interface index for it.

Using UUID instead of interface name  in `NM.RemoteConnection.props.master`
, `NM.SettingVlan.props.parent` and etc could solve this problem.

Integration test cases included.

Signed-off-by: Gris Ge <fge@redhat.com>
